pkg/profilecreator/profilecreator.go

```diff
@@ -231,3 +233,64 @@ func (ghwHandler GHWHandler) CPU() (*cpu.Info, error) {
 func (ghwHandler GHWHandler) Topology() (*topology.Info, error) {
 	return ghw.Topology(ghwHandler.snapShotOptions)
 }
+
+// GetReservedAndIsolatedCPUs returns Reserved and Isolated CPUs
+func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, splitReservedCPUsAcrossNUMA bool) (string, string, error) {
+	if reservedCPUCount < 0 {
+		return "", "", fmt.Errorf("Specified eservered CPU count is negative, please specify it correctly")
+	}
+	topologyInfo, err := ghwHandler.Topology()
+	if err != nil {
+		return "", "", fmt.Errorf("Error obtaining Topology Info from GHW snapshot: %v", err)
+	}
+	reservedCPUSet := cpuset.NewBuilder()
+	isolatedCPUSet := cpuset.NewBuilder()
+	var numaNodeNum int
+	if splitReservedCPUsAcrossNUMA {
+		numaNodeNum = len(topologyInfo.Nodes)
+	} else {
+		numaNodeNum = 1
+	}
+
+	var max = 0
+	reservedPerNuma := reservedCPUCount / numaNodeNum
+	remainder := reservedCPUCount % numaNodeNum
+	for numaID, node := range topologyInfo.Nodes {
+		if splitReservedCPUsAcrossNUMA {
+			if remainder != 0 {
+				max = (numaID+1)*reservedPerNuma + (numaNodeNum - remainder)
+				remainder--
+			} else {
+				max = max + reservedPerNuma
+			}
+		} else {
+			max = reservedCPUCount
+		}
+		//sorting by comparing the first logical core of processor cores
+		sort.Slice(node.Cores, func(i, j int) bool {
+			//sorting the logical cores a processor
+			sort.Slice(node.Cores[i].LogicalProcessors, func(x, y int) bool {
+				return node.Cores[i].LogicalProcessors[x] < node.Cores[i].LogicalProcessors[y]
+			})
+			sort.Slice(node.Cores[j].LogicalProcessors, func(x, y int) bool {
+				return node.Cores[j].LogicalProcessors[x] < node.Cores[j].LogicalProcessors[y]
+			})
```

Comment:
three sorts look suspicious. Can we squash the two inner sorts together somehow?

Reply:
I can try but the idea is that we first individually sort the Logical Processors (hence the two inner sort) based on CPUID and then then sort the cores by comparing the smallest logical core ID so I think we need the two inner sort but I can try to figure out better ways of doing this.